test/inference/test_inference.cpp

```diff
@@ -36,3 +38,35 @@ TEST(bfs, check_result_vec) {
   std::vector<int> res = {81, 81, 81};
   ASSERT_EQ(tmp, res);
 }
+TEST(bfs, check_e2e) {
```

Comment:
Need to come up with better name here. Out of the context it is not clear what is considered end-to-end here. And even it looks similar to the previous test which is not considered e2e